chemprop/uncertainty/evaluator.py

```diff
@@ -47,21 +46,46 @@ class CalibrationAreaEvaluator(RegressionEvaluator):
     """
     A class for evaluating regression uncertainty values based on how they deviate from perfect
     calibration on an observed-probability versus expected-probability plot.
+
+    Parameters
+    ----------
+    preds: Tensor
+        the predictions for regression tasks. It is a tensor of the shape of ``n x t``, where ``n`` is the number of input
+        molecules/reactions, and ``t`` is the number of tasks.
+    uncs: Tensor
+        the predicted uncertainties (variance) of the shape of ``n x t``
+    targets: Tensor
+        a tensor of the shape ``n x t``
+    mask: Tensor
+        a tensor of the shape ``n x t`` indicating whether the given values should be used in the evaluation
+    num_bins: int
+        the number of bins to discretize the [0, 1] interval
+
+    Returns
+    -------
+    Tensor
+        a tensor of the shape ``t`` containing the evaluated metrics
     """
 
-    def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        bin_scaling = torch.tensor([erfinv(i / 100) * np.sqrt(2) for i in range(1, 100)])
-        bin_scaling = bin_scaling.view(-1, 1, 1)
+    def evaluate(
+        self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor, num_bins: int = 100
+    ) -> Tensor:
+        bin_scaling = torch.special.erfinv(torch.arange(1, num_bins) / num_bins).view(
+            -1, 1, 1
+        ) * np.sqrt(2)
         errors = torch.abs(preds - targets)
         uncs = torch.sqrt(uncs).unsqueeze(0)
         bin_unc = uncs * bin_scaling
         bin_count = bin_unc >= errors.unsqueeze(0)
         mask = mask.unsqueeze(0)
-        bin_fraction = (bin_count & mask).sum(1) / mask.sum(1)
-        fractions = torch.zeros((uncs.shape[-1], 101))
-        fractions[:, 100] = 1.0
-        fractions[:, 1:100] = bin_fraction.permute(1, 0)
-        miscal_area = torch.sum(0.01 * torch.abs(fractions - torch.arange(101) / 100), dim=1)
+        bin_fractions = (bin_count & mask).sum(1) / mask.sum(1)
+        num_tasks = uncs.shape[-1]
+        bin_fractions = torch.cat(
+            [torch.zeros(1, num_tasks), bin_fractions, torch.ones(1, num_tasks)]
+        ).T
+        miscal_area = torch.sum(
+            0.01 * torch.abs(bin_fractions - torch.arange(num_bins + 1) / num_bins), dim=1
+        )
```

Comment:
```suggestion
        observed_auc = (bin_count & mask).sum(1) / mask.sum(1)
        num_tasks = uncs.shape[-1]
        observed_auc = torch.cat(
            [torch.zeros(1, num_tasks), observed_auc, torch.ones(1, num_tasks)]
        ).T
        ideal_auc = torch.arange(num_bins + 1) / num_bins
        miscal_area = 0.01 * (observed_auc - ideal_auc).abs().sum(dim=1)
```
also, where's `0.01` come from? Is it just `1 / num_bins`?

Reply:
Good catch! That is `1 / num_bins`.